### PR TITLE
Chore: Add a build

### LINKED ISSUE #16: Create a build
Currently there isn't an actual build. The src is published as is and compositions can only be used in typescript projects. Need to create a dist package for native js

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Closes #16

### CODE REVIEW

File: vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  resolve: {
+    alias: [
+      {
+        find: '@',
+        replacement: resolve(__dirname, './src'),
+      },
+    ],
+  },
+  build: {
+    emptyOutDir: false,
```
Comment (znicholasbrown) [inline vite.config.ts:14]: I think you can omit this field (it defaults to true) unless there's a reason you wouldn't want to remove files in the `/dist` folder whenever you're making a build?

Comment (znicholasbrown) [reply vite.config.ts:14]: Ah just kidding, that's so your type files aren't overwritten

Comment (pleek91) [reply vite.config.ts:14]: That's what we're doing in the other projects. I remember we added it for some reason. But I don't remember why. Seems like we shouldn't need it though 🤔 I'll remove.

Comment (znicholasbrown) [reply vite.config.ts]: No we for sure need it since the `tsc` command you're chaining puts files in `/dist/types`

Comment (pleek91) [reply vite.config.ts:14]: jk its back! 😂

**Review by znicholasbrown — APPROVED** [review]

**Comment by znicholasbrown** [comment]
Looks good to me @pleek91

**Pull request merged**